sdk/core/core-amqp/src/errors.ts

```diff
@@ -639,12 +639,15 @@ const rheaPromiseErrors = [
  * @param err - The amqp error that was received.
  * @returns MessagingError object.
  */
-export function translate(err: AmqpError | Error): MessagingError | Error {
+export function translate(err: unknown): MessagingError | Error {
   if (!isDefined(err)) {
     return new Error(`Unknown error encountered.`);
   } else if (typeof err !== "object") {
     // The error is a scalar type, make it the message of an actual error.
-    return new Error(err);
+    return new Error(String(err));
+  }
+  if (!(err instanceof Error) && !isAmqpError(err)) {
```

Comment:
use `isError` from core-util?

Reply:
core-amqp doesn't depend on core-util, and I am a bit hesitant to add it as core-amqp is in its own messaging space...

Reply:
I'll leave it up to you, though I would hope we would keep core-util general enough that it can be shared across messaging / rest / grpc / whatever else

Reply:
now looking at removing `: any` I think I will do that in a later PR.